### ISSUE TITLE
Layout/MultilineMethodCallIndentation does not enfore indentation inside arguments

### ISSUE THREAD

**Issue by jviney** [rubocop/rubocop#10059]
Layout/MultilineMethodCallIndentation does not enforce indentation inside method arguments:

```ruby
bar(
  baz
  .not_indented
)
```

## Describe the solution you'd like

The rule should be triggered by this code example.

**Comment by sl4vr** [comment]
It exists in specs so it might be a desired behaviour.
https://github.com/rubocop/rubocop/blob/95fd7b70eb7da241d14044ecbbf5d0c1496dec91/spec/rubocop/cop/layout/multiline_method_call_indentation_spec.rb#L883-L889

**Comment by jviney** [comment]
Interesting. Does anyone know why that's the case? Maybe there could be a new option to enforce the rule inside method parameters.

**Comment by jonas054** [comment]
I was the one who added `MultilineOperationIndentation`, which at that time also had the responsibility to check method calls. Later on, when `MultilineMethodCallIndentation` had been extracted, the issue #3969 was reported and fixed, which led to indentation of method calls that were parameters to other method calls - as long as there were no parentheses.

Unfortunately, I can't recall why operations or method calls inside the parentheses of a method call were excluded from inspection. But I agree that a new configuration parameter would be a good way to modify the cop at this point. Current behavior would have to be the default.

**Comment by IslamElsayed** [comment]
I'd like to take this, with one scoping question first.

Still reproduces on current master (`7b1e12b`), with `EnforcedStyle: indented`:

```ruby
bar(
  baz
.not_indented      # no offense
)

qux = baz
.also_not_indented # offense: "Align .also_not_indented with baz on line 6"
```

Same construct, flagged outside the parentheses and ignored inside them. The cause is `not_for_this_cop?` in `MultilineExpressionIndentation`:

```ruby
def not_for_this_cop?(node)
  node.ancestors.any? do |ancestor|
    grouped_expression?(ancestor) || inside_arg_list_parentheses?(node, ancestor)
  end
end
```

`inside_arg_list_parentheses?` returns true for any node positioned between a parenthesized send's `begin` and `end`, so the whole subtree is skipped regardless of how it is indented.

@sl4vr's point above is worth taking seriously — the behaviour is pinned by specs, so it isn't accidental in the sense of "no one noticed". But @jonas054, who wrote the cop, said he can't recall why parenthesized calls were excluded and that **"a new configuration parameter would be a good way to modify the cop at this point. Current behavior would have to be the default."** That's what I'd implement: an opt-in parameter, default off, with the existing specs left passing unchanged as the proof that the default is untouched.

**The scoping question.** `not_for_this_cop?` lives in the shared `MultilineExpressionIndentation` mixin and has exactly two callers:

- `Layout/MultilineMethodCallIndentation` (this issue)
- `Layout/MultilineOperationIndentation`

So the parameter can go in one of two places, and they are not equivalent:

1. **On `MultilineMethodCallIndentation` only** — narrower, matches the issue as filed. The two cops then disagree about whether parentheses suppress inspection, which is a bit odd given they share the mechanism.
2. **On both cops** — consistent, and closer to the original arrangement you described, where `MultilineOperationIndentation` also handled method calls. But it is a larger surface and changes a cop nobody has complained about.

I lean towards (1) — fix what was reported, don't widen a 4-year-old issue on my own initiative — but you have the history here and I'd rather ask than guess. Happy to do (2) if you would rather the pair stay symmetrical.

Also, is there a name you'd prefer for the parameter? `AllowInsideParenthesizedArgumentList` (default `true`) keeps the default readable as "current behavior", but I don't want to invent config vocabulary that cuts against existing conventions.

Disclosure: I use AI assistance when working on patches, and I check and test everything before submitting.
